axelrod/strategy_transformers.py

```diff
@@ -309,9 +350,46 @@ def mixed_wrapper(player, opponent, action, probability, m_player):
 
     return action
 
+
 MixedTransformer = StrategyTransformerFactory(
     mixed_wrapper, name_prefix="Mutated")
 
+
+def joss_ann_wrapper(player, opponent, proposed_action, probability):
+    """Wraps the players strategy function to produce the Joss-Ann.
+
+    The Joss-Ann of a strategy is a new strategy which has a probability of
+    choosing the move C, a probability of choosing the move D, and otherwise
+    uses the response appropriate to the original strategy.
+
+    Parameters
+    ----------
+
+    player: Player object or subclass (self)
+    opponent: Player object or subclass
+    proposed_action: axelrod.Action, C or D
+        The proposed action by the wrapped strategy
+    probability: tuple
+        a tuple or list representing a probability distribution of playing move
+        C or D (doesn't have to be complete) ie. (0, 1) or (0.2, 0.3)
+
+    Returns
+    -------
+    action: an axelrod.Action, C or D
+    """
+    if sum(probability) > 1:
+        probability[:] = [i / sum(probability) for i in probability]
+
+    remaining_probability = max(0, 1 - probability[0] - probability[1])
+    probability += (remaining_probability,)
+    options = [C, D, proposed_action]
+    action = choice(options, p=probability)
+    return action
```

Comment:
Could we not implement this as a particular case of the Mixed Strategy transformer (by mixing between the probe and Cooperator and Defector)? This would reduce duplication.